sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -772,6 +772,8 @@ object SQLConf {
         " produced by a builtin function such as to_char or CAST")
       .version("4.0.0")
       .stringConf
+      .checkValue(CollationFactory.collationNameToId(_) >= 0,
```

Comment:
There is an overload of `checkValue` that takes error class name and parameters. We should use it instead of throwing exception in the check function.

Reply:
The reason why I left it as it is, is that we give suggestion for close name. I am not sure how would we pass the suggestion if done here.

Reply:
can we change the required parameter type from `parameters: Map[String, String]` to `parameters: T => Map[String, String]`? Then we can generate the error message with the actual config value.